tests/daemon/test-circular-file.c

```diff
@@ -609,11 +611,158 @@ test_circular_file_shrink (Fixture      *fixture,
   g_object_unref (circular_file_2);
 }
 
+static void
+test_circular_file_broken_metadata_file (Fixture      *fixture,
+                                         const gchar  *metadata_file_contents,
+                                         gssize        metadata_file_length,
+                                         GKeyFileError expected_error_code)
+{
+  const gsize max_size = 1024;
+  g_autofree gchar *data_file_contents = g_malloc (max_size);
+  g_autofree gchar *metadata_file_path =
+    g_strconcat (fixture->data_file_path, METADATA_EXTENSION, NULL);
+  g_autoptr(GError) error = NULL;
+  gboolean ret;
+
+  /* Fill the data file with junk. */
+  memset (data_file_contents, '*', max_size);
+  ret = g_file_set_contents (fixture->data_file_path, data_file_contents, max_size, &error);
+  g_assert_no_error (error);
+  g_assert_true (ret);
+
+  /* Initialize the metadata file to the desired incorrect contents */
+  ret = g_file_set_contents (metadata_file_path, metadata_file_contents, metadata_file_length, &error);
+  g_assert_no_error (error);
+  g_assert_true (ret);
+
+  g_autoptr(EmerCircularFile) circular_file = NULL;
+
+  circular_file = emer_circular_file_new (fixture->data_file_path, max_size,
+                                          /* reinitialize */ FALSE,
+                                          &error);
+  g_assert_null (circular_file);
+  g_assert_error (error, G_KEY_FILE_ERROR, expected_error_code);
+  g_clear_error (&error);
+
+  circular_file = emer_circular_file_new (fixture->data_file_path, max_size,
+                                          /* reinitialize */ TRUE,
+                                          &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (circular_file);
+
+  /* Verify that the file is logically empty, even though it's physically full
+   * of asterisks.
+   */
+  const gchar * const NO_STRINGS[0];
+  guint64 token = assert_strings_read (circular_file, NO_STRINGS, 0);
```

Comment:
I know this is not your code, and also it could be due to me being used to other naming conventions, but I found it surprisingly confusing that `assert_strings_read()` does actually more stuff than simply asserting the desired outcome/state. It would be more clear, IMHO, if it was named something like `read_strings_and_check()` or similar, so that it's clear that it performs both the action that needs to be tested + the actual test.

Not for this patch though, but maybe to consider for a separate cleanup commit?

Reply:
Yes, great idea.